include/volume/volume_cooling_gaussians.hpp

```diff
@@ -22,28 +22,47 @@
 #include "random_walks/gaussian_helpers.hpp"
 #include "random_walks/gaussian_ball_walk.hpp"
 #include "random_walks/gaussian_cdhr_walk.hpp"
+#include "random_walks/gaussian_hamiltonian_monte_carlo_exact_walk.hpp"
 #include "sampling/random_point_generators.hpp"
 #include "volume/math_helpers.hpp"
 
+typedef long double LNT;
 
 /////////////////// Helpers for random walks
 
 template <typename WalkType>
 struct update_delta
 {
+
     template <typename NT>
     static void apply(WalkType, NT) {}
+    /*	
+    template <typename LNT>
+    static void apply(WalkType walk, LNT delta) {
+	    walk.update_delta(delta);
+    }
+    */
 };
 
 template <typename Polytope, typename RandomNumberGenerator>
-struct update_delta<GaussianBallWalk::Walk<Polytope, RandomNumberGenerator>>
+//struct update_delta<GaussianBallWalk::Walk<Polytope, RandomNumberGenerator>>
```

Comment:
why removing the specialization of `GaussianBallWalk::Walk<Polytope, RandomNumberGenerator>`? 
You can keep both.